File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        i, j =0,1
        mprof = 0
        while j <len(prices):
            prof = prices[j] - prices[i]
            mprof = max(prof, mprof)
            if prof < 0:
                i =j
            j+=1
        return mprof
            

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int buy = prices[0];
        int res =0;
        for (int i =1;i< prices.size();i++){
            int sell = prices[i];
            if (buy < sell) res = max(res,sell-buy);
            else buy = sell;
        }
      return res;  
    }
};